src/completion.ts

```diff
@@ -241,6 +243,17 @@ class DictionaryCompletionItemProvider implements vscode.CompletionItemProvider
                     return [];
                 }
                 return this.completeByFirstLetter(firstLetter, addSpace);
+            case "quarto":
+                // [caption](don't complete here)
+                if (/\[[^\]]*\]\([^\)]*$/.test(textBefore)) {
+                    return [];
+                }
+                // ```{lang}``` -- code blocks TODO: not working, new to regexp
```

Comment:
You can use https://regex101.com/ to help you debug the regexp (remember to select ECMAScript/JavaScript in the left).

And you can remove this TODO if it works as expected now 😉